pkg/controllers/provisioning/v1alpha1/reallocation/terminate.go

```diff
@@ -19,21 +19,125 @@ import (
 	"fmt"
 	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
 	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/utils/functional"
+	utilsnode "github.com/awslabs/karpenter/pkg/utils/node"
+	"github.com/awslabs/karpenter/pkg/utils/ptr"
+	"github.com/awslabs/karpenter/pkg/utils/scheduling"
 	"go.uber.org/zap"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type Terminator struct {
 	kubeClient    client.Client
 	cloudprovider cloudprovider.Factory
+	coreV1Client  corev1.CoreV1Interface
 }
 
-// DeleteNodes will use a cloudprovider implementation to delete a set of nodes
-func (t *Terminator) DeleteNodes(ctx context.Context, nodes []*v1.Node, spec *v1alpha1.ProvisionerSpec) error {
+func (t *Terminator) ManageTermination(ctx context.Context, provisioner *v1alpha1.Provisioner) error {
+	// 1. Cordon each terminable node
+	if err := t.cordonNodes(ctx, provisioner); err != nil {
+		return fmt.Errorf("cordoning expired nodes, %w", err)
```

Comment:
I think this annotation may not age well if we decide to make nodes terminable in other ways via expiry. Consider cordoning terminable node.